### PR TITLE
Madiha Js-core-3

### PR DESCRIPTION
**Pull request by diya2214**

### CODE REVIEW

File: I.write-tests/03-remove-vowels-from-array/remove-vowels-in-array.test.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
+const removeVowelsForWords = require('./remove-vowels-in-array')
 test('remove vowels from all words in array', function () {
-    // Arrange
-
-    // Act
-
-    // Assert
+  // Arrange
+  const myArray = ['Irina', 'Etza', 'Daniel']
+  const expectation = ['rn', 'tz', 'dnl']
```
Comment (sansaid) [inline I.write-tests/03-remove-vowels-from-array/remove-vowels-in-array.test.js]: Are you sure this is the correct expecation? Does the removeVowels function also lowercases the constenants?

File: II.pass-tests/05-factorial/factorial.js [hunk 2]
```diff
@@ -8,8 +8,13 @@
 // calculate and return the factorial of int
 // note: factorial of 0 is 1
 
-function factorial(int) {
-
+function factorial (int) {
+  if (int >= 0) {
+    var fact = 1
+    for (i = 2; i <= int; i++) {
+      fact = fact * i
```
Comment (sansaid) [inline II.pass-tests/05-factorial/factorial.js]: You could make this more concise with fact =* i (this just translates to 'mutiply myself with i')

Comment (sansaid) [inline II.pass-tests/05-factorial/factorial.js]: You can make this line more concise by changing to `fact *= i` (this just translates to 'multiply myself with i') - you can even replace this with any operator (+, -, /) and it'll do the same thing for that operation

File: I.write-tests/01-greet-people/greet-people.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
-function greetPeople(people) {
-    var greeting = 'Hello ';
+function greetPeople (people) {
+  var greeting = 'Hello '
+  return greeting + people.join(', ');
```
Comment (lpalma) [inline I.write-tests/01-greet-people/greet-people.js]: Well done for using `join` :clap: 
Do we need the comments below? If they are not adding value to the code, we should remove it :)

File: II.pass-tests/03-sum-numbers/add-numbers.js [hunk 4]
```diff
@@ -1,4 +1,7 @@
-function addNumbers(numbers) {
-
+function addNumbers (numbers) {
+  return numbers.reduce(
```
Comment (lpalma) [inline II.pass-tests/03-sum-numbers/add-numbers.js]: excellent :smiley: 
`reduce` is one of my favourite functions

File: II.pass-tests/10-paint-ford/paint-cars.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const paintShop = function (cars, newColour) {
+  // var paintedCars = cars.slice()
+  var paintedCars = JSON.parse(JSON.stringify(cars))// duplicating array without a refrence to original array
+  paintedCars[0].colour = newColour
```
Comment (lpalma) [inline II.pass-tests/10-paint-ford/paint-cars.js]: can we guarantee that the car to be painted will always be the first element?
How can we make this function more _reusable_? :)

File: I.write-tests/01-greet-people/greet-people.js (hunk 3 above)
Comment (diya2214) [reply I.write-tests/01-greet-people/greet-people.js]: noted thanks

File: II.pass-tests/03-sum-numbers/add-numbers.js (hunk 4 above)
Comment (diya2214) [reply II.pass-tests/03-sum-numbers/add-numbers.js]: mine too it makes things so easy but it took me time to get my head around it :)

File: II.pass-tests/10-paint-ford/paint-cars.js (hunk 5 above)
Comment (diya2214) [reply II.pass-tests/10-paint-ford/paint-cars.js]: hmm I wrote the code based on test already written ... will discuss this with you in class

File: II.pass-tests/05-factorial/factorial.js (hunk 2 above)
Comment (diya2214) [reply II.pass-tests/05-factorial/factorial.js]: noted already started using it

File: I.write-tests/03-remove-vowels-from-array/remove-vowels-in-array.test.js (hunk 1 above)
Comment (diya2214) [reply I.write-tests/03-remove-vowels-from-array/remove-vowels-in-array.test.js]: oppps corrected thanks

**Pull request closed without merging**